src/som/compiler/Parser.java

```diff
@@ -1444,6 +1450,31 @@ private LiteralNode literalString() throws ParseError {
     return new StringLiteralNode(s, getSource(coord));
   }
 
+  private LiteralNode literalArray(final MethodBuilder builder) throws ProgramDefinitionError {
+    SourceCoordinate coord = getCoordinate();
+    List<ExpressionNode> expressions = new ArrayList<ExpressionNode>();
+
+    expect(NewLiteralArray, DelimiterOpeningTag.class);
+
+    boolean sawPeriod = true;
+
+    while (true) {
+      comments();
+
+      if (sym == EndLiteralArray) {
+        expect(EndLiteralArray, DelimiterClosingTag.class);
+        return createLiteralArray(expressions, getSource(coord));
+      }
+
+      if (!sawPeriod) {
+        expect(Period, null);
+      }
+
+      expressions.add(expression(builder));
+      sawPeriod = accept(Period, StatementSeparatorTag.class);
+    }
```

Comment:
I restructured the while loop, which is now more concise using a switch statement. The current implementation expects the user to properly provide expressions, with periods delimiting them (with the last being optional). 

Currently there is nothing done to communicate why the program's definition is invalid. Should I add something to accompany the `Program Definition Error`in cases where a parsable expression cannot be found (or will this be handled sufficiently in parse expression)?